File: homework4_2/src/main/java/com/example/homework4_2/MainActivity.java
package com.example.homework4_2;

import android.os.Bundle;
import android.widget.CompoundButton;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SwitchCompat;

import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity implements CoordinateListener {
    private Custom custom;
    private SwitchCompat switchCompat;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        custom = findViewById(R.id.customView);
        custom.setCoordinateListener(this);

        switchCompat = findViewById(R.id.swither);
        switchCompat.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                if (isChecked) {
                    Toast.makeText(MainActivity.this, "Включен режим Toast", Toast.LENGTH_SHORT).show();
                } else {
                    Snackbar.make(custom, "Включен режим Snackbar", Snackbar.LENGTH_SHORT).show();
                }
            }
        });
    }

    @Override
    public void getCoordinates(float x, float y, int color) {
        if (switchCompat.isChecked()) {
            Toast.makeText(this, "Координаты [" + Math.round(x) + ";" + Math.round(y) + "]", Toast.LENGTH_SHORT).show();
        } else {
            Snackbar snackbar = Snackbar.make(custom, "Координаты [" + Math.round(x) + ";" + Math.round(y) + "]", Snackbar.LENGTH_SHORT);
            snackbar.getView().setBackgroundColor(color);
            snackbar.show();
        }
    }
}

File: homework10/src/main/java/ru/academy/homework10/MainActivity.kt
package ru.academy.homework10

import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.content.SharedPreferences
import android.net.ConnectivityManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.CompoundButton
import android.widget.Switch
import android.widget.Toast
import androidx.appcompat.widget.SwitchCompat

class MainActivity : AppCompatActivity() {

    private val broadcastReceiver by lazy { EventBroadcastReceiver() }
    private var b: Boolean = false
    private lateinit var switch: SwitchCompat

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.stopService).setOnClickListener {
            stopService(Intent(this@MainActivity, ServiceExample::class.java))
        }

        switch = findViewById<SwitchCompat>(R.id.switcher)
        switch.isChecked = loadBooleanStatement()

        switch.setOnCheckedChangeListener { _, isChecked ->
            if (isChecked) {
                Toast.makeText(this, "External Storage", Toast.LENGTH_SHORT).show()
                b = true
                saveBooleanStatement(b)
            } else {
                Toast.makeText(this, "Internal Storage", Toast.LENGTH_SHORT).show()
                b = false
                saveBooleanStatement(b)
            }
        }
    }

    override fun onStart() {
        super.onStart()
        findViewById<Button>(R.id.startService).setOnClickListener {
            val intentFilter = IntentFilter().apply {
                addAction(Intent.ACTION_TIME_CHANGED)
                addAction(Intent.ACTION_TIME_TICK)
                addAction(Intent.ACTION_TIMEZONE_CHANGED)
            }
            registerReceiver(broadcastReceiver, intentFilter)
            startService(Intent(this@MainActivity, ServiceExample::class.java))
        }
    }

    private fun saveBooleanStatement(b: Boolean) {
        val sharedPreferences = getSharedPreferences("PREF", Context.MODE_PRIVATE)
        var ed: SharedPreferences.Editor = sharedPreferences.edit()
        ed.putBoolean("STATEMENT", b)
        ed.apply()
    }

    private fun loadBooleanStatement(): Boolean {
        val sharedPreferences = getSharedPreferences("PREF", Context.MODE_PRIVATE)
        return sharedPreferences.getBoolean("STATEMENT", false)
    }
}
